Add Record Type mapping for Bulk Data tasks

davidmreed opened the pull request:
# Critical Changes

# Changes

- Add support for mapping Record Types between orgs (by Developer Name) during bulk data extract and load.
- Add support for Record Type mapping in bulk mapping generator.
- Corrects a problem resulting in `generate_dataset_mapping` outputting an incorrectly sequenced mapping.
- Adds new documentation for bulk data tasks

# Issues Closed

- Fixes #1270

jstvz commented:
Unrelated to your changes: Do we still need `unicodecsv`?

davisagli commented:
@jstvz good question -- probably not

davidmreed commented:
I don't think so. I've used Python 3 `csv` with Unicode data extensively.

Br4nd0R commented on cumulusci/tasks/bulkdata/utils.py:
```diff
@@ -128,6 +130,29 @@ def _sql_bulk_insert_from_csv(self, conn, table, columns, data_file):
         self.session.flush()
 
+    def _create_record_type_table(self, table_name):
+        rt_map_model_name = "{}Model".format(table_name)
```
```suggestion
        rt_map_model_name = f"{table_name}Model"
```

Br4nd0R commented on cumulusci/tasks/bulkdata/utils.py:
```diff
@@ -128,6 +130,29 @@ def _sql_bulk_insert_from_csv(self, conn, table, columns, data_file):
+        mapper(self.models[table_name], rt_map_table)
+
+    def _extract_record_types(self, sobject, table, conn):
+        self.logger.info("Extracting Record Types for {}".format(sobject))
```
```suggestion
        self.logger.info(f"Extracting Record Types for {sobject}")
```

davidmreed commented:
@Br4nd0R I converted the bulk data module to fstrings throughout, added documentation, and addresseed @prescod's suggestion that the mapping generator should refrain from lower-casing Salesforce API names.
